code/mob/living/silicon/ai/holograms.dm

```diff
@@ -0,0 +1,71 @@
+//AI HOLOGRAMS
+/mob/living/silicon/ai/proc/create_hologram()
+	if (!deployed_to_eyecam)
+		boutput(src, "Deploy to an AI Eye first to create a hologram.")
+		return
+	if (islist(holograms) && holograms.len >= max_holograms)
+		boutput(eyecam, "Not enough RAM to project more holograms. Delete others to make room.")
+		return
+	var/turf/T = get_turf(src.eyecam)
+	if (!istype(T) || !istype(T.cameras) || T.cameras.len == 0)
```

Comment:
```suggestion
	if (!istype(T) || !istype(T.cameras) || !length(T.cameras))
```
somehow this change didn't make it into my review...
Anyways, `length` is faster than `.len`